### PR TITLE
nuevas actualizaciones en la edicion del perfil

### PR DESCRIPTION
**Pull request by ervinj1998**
En esta actualizacion de la edicion del perfil, se establecio condiciones al usuario con los datos ingresados como por ejemplo: 
 - [x] El nombre no puede estar en blanco
 - [x] El emain tiene que cumplicar con la sintaxis correcta "email@dominio.com".
  -[x] El password debe de ser al menos 6 caracteres
se agregi mensajes de alerta para estas tres opciones en caso de que sucedan.

### CODE REVIEW

**Comment by Hawaiideveloper** [comment]
In this update of the profile edition, conditions were established for the user with the data entered, such as:

 - [  ]  The name cannot be blank
 - [  ]  The email has to comply with the correct syntax "email@domain.com".
- [x] The password must be at least 6 characters
Alert messages were added for these three options in case they happen.

File: src/user/EditProfile.js [hunk 1]
```diff
@@ -39,6 +40,25 @@ componentDidMount() {
...
 }
 
+isValid = () => {
+    const {name, email, password} = this.state
+    if(name.length == 0){
+        this.setState({error: "Name is requerid"}) //validamos y condicionamos que el nombre no puede estar en blanco si quiere editar el perfil.
+        return false
+    }
+        //  email@domain.com
+    if(!/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email)){ //evalua si se esta ingresando un email con las caracteristicas correctas
```
Comment (Hawaiideveloper) [inline src/user/EditProfile.js:50]: regex

File: src/user/EditProfile.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ export class EditProfile extends Component { //componente para habilitar el boto
             name: "",
             email: "",
             password: "",
-            redirectToProfile: false
+            redirectToProfile: false,
+            error: ""
```
Comment (Hawaiideveloper) [inline src/user/EditProfile.js:15]: we added an error to this field

File: src/user/EditProfile.js [hunk 3]
```diff
@@ -53,8 +73,11 @@ handleChange = (name) => (event) => { //higher order function returns another fu
 }
 
 clickSubmit = event => {
-    event.preventDefault()
-    const { name, email, password } = this.state //destructure the word "this"
+    event.preventDefault();
+
+    if(this.isValid()) { // dependiendo de la validacion se ejecutara lo siguiente
```
Comment (Hawaiideveloper) [inline src/user/EditProfile.js:78]: great conditional statement

**Comment by Hawaiideveloper** [comment]
I will wait for the next two and then the last one, will be ran and tested, after this one.  

 But our goal should be to finish at least 6 per day

File: src/user/EditProfile.js (hunk 1 above)
Comment (ervinj1998) [reply src/user/EditProfile.js:50]: Yes

**Comment by Hawaiideveloper** [comment]
let me know when the next 2 are done as we want to do them in 3s

**Comment by Hawaiideveloper** [comment]
now merging into alpha for testing

**Pull request merged**